go.mod

```diff
@@ -1,5 +1,8 @@
 module github.com/edgexfoundry/go-mod-secrets
 
-require github.com/stretchr/testify v1.3.0
+require (
+	github.com/edgexfoundry/go-mod-core-contracts v0.1.42
```

Comment:
New dependency to facilitate usage of logger.LoggingClient abstraction.

@lenny-intel called this out in a prior review:

> Modules typically don't log message directly. If EdgeX community is ok with passing in a reference to the logger so you can log the message, then I good with that.

I'm not OK with this.  You're taking a dependency on an external module that adds coupling and doesn't provide any real value.  Instead, I suggest you create your own local interface:

```
type loggingClient interface {
	// Debug logs a message at the DEBUG severity level
	Debug(msg string, args ...interface{})
	// Info logs a message at the INFO severity level
	Info(msg string, args ...interface{})
	// Warn logs a message at the WARN severity level
	Warn(msg string, args ...interface{})
}
```

Accept a concrete implementation of `loggingClient` within your constructor.  The existing `logger.LoggingClient` will satisfy that contract.  This avoids taking the unnecessary module dependency.

This also lets the module's user determine whether or not it wants to share its log with you.  It may be that it wants to ignore your logging messages completely (in which case it can provide a null object implementation of the contract).  You could argue the same thing could be done with the go-mod-core-contracts abstraction -- and it can.  However, when you use the go-mod-core-contracts abstraction, it's more likely you're going to get someone blindly passing along the existing logger -- which isn't necessarily desirable in all cases.

Reply:
good suggestion. can do that

Reply:
Done

Reply:
The dependency should be removed.